spec/requests/filter_parameter_logging_spec.rb

```diff
@@ -0,0 +1,83 @@
+require "rails_helper"
+
+RSpec.describe "Filter parameter logging spec", type: :request do
+  let(:request_user) { FactoryBot.create(:user) }
+  let(:sync_route) { "/api/v3/patients/sync" }
+  let(:build_payload) { -> { build_patient_payload(FactoryBot.build(:patient, registration_facility: request_user.facility)) } }
+
+  let(:auth_headers) do
+    {"HTTP_X_USER_ID" => request_user.id,
+     "HTTP_X_FACILITY_ID" => request_user.facility.id,
+     "HTTP_AUTHORIZATION" => "Bearer #{request_user.access_token}"}
+  end
+  let(:headers) do
+    {"ACCEPT" => "application/json", "CONTENT_TYPE" => "application/json"}.merge(auth_headers)
+  end
+
+  it "does not log sensitive patient data" do
+    stringio = StringIO.new
+    test_logger = Logger.new(stringio)
+    allow(ActionController::Base).to receive(:logger).and_return(test_logger)
+
+    patient_payloads = (1..3).map { build_payload.call }
+    post sync_route, params: {patients: patient_payloads}.to_json, headers: headers
+    get sync_route, params: {}, headers: headers
+
+    patients = patient_payloads.map { |payload| Patient.find(payload[:id]) }
+    output = stringio.string
+    patients.each do |patient|
+      expect(output).to_not match(/\b#{patient.full_name}\b/)
+      expect(output).to_not match(/\b#{patient.age}\b/) if patient.age
+      expect(output).to_not match(/\b#{patient.date_of_birth}\b/) if patient.date_of_birth
+      if patient.address
```

Comment:
We should filter all address fields here, including district and state.

We should also filter phone numbers.

Reply:
done